Sources/IO/Core/HttpDataSetSeriesReader/index.js

```diff
@@ -0,0 +1,214 @@
+    (error) => {
+      reject(error);
+    }
+  );
+}
+
+// ----------------------------------------------------------------------------
+// vtkHttpDataSetSeriesReader methods
+// ----------------------------------------------------------------------------
+
+function vtkHttpDataSetSeriesReader(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkHttpDataSetSeriesReader');
+
+  // Empty output by default
+  model.output[0] = vtk({ vtkClass: 'vtkPolyData' });
+
+  // Create default dataAccessHelper if not available
+  if (!model.dataAccessHelper) {
+    model.dataAccessHelper = HTTP_DATA_ACCESS;
+  }
+
+  model.currentReader = null;
+
+  publicAPI.updateMetaData = (loadData = false) => {
+    if (model.compression === 'zip') {
+      return new Promise((resolve, reject) => {
+        HTTP_DATA_ACCESS.fetchBinary(model.baseURL).then((zipContent) => {
+          model.dataAccessHelper = DataAccessHelper.get('zip', {
+            zipContent,
+            callback: (zip) => {
+              model.baseURL = '';
+              model.dataAccessHelper.fetchJSON(publicAPI, 'index.json').then(
+                (dataset) => {
+                  processDataSet(
+                    publicAPI,
+                    model,
+                    dataset,
+                    resolve,
+                    reject,
+                    loadData
+                  );
+                },
+                (error) => {
+                  reject(error);
+                }
+              );
+            },
+          });
+        });
+      });
+    }
+
+    return new Promise((resolve, reject) => {
+      model.dataAccessHelper.fetchJSON(publicAPI, model.url).then(
+        (dataset) => {
+          return processDataSet(
+            publicAPI,
+            model,
+            dataset,
+            resolve,
+            reject,
+            loadData
+          );
+        },
+        (error) => {
+          reject(error);
+        }
+      );
+    });
+  };
+
+  publicAPI.setUrl = (url, options = {}) => {
+    if (url.indexOf('index.json') === -1 && !options.fullpath) {
+      model.baseURL = url;
+      model.url = `${url}/index.json`;
+    } else {
+      model.url = url;
+
+      // Remove the file in the URL
+      const path = url.split('/');
+      path.pop();
+      model.baseURL = path.join('/');
+    }
+
+    model.compression = options.compression;
+
+    return publicAPI.updateMetaData(!!options.loadData);
+  };
+
+  publicAPI.getTimeSteps = () => {
+    const timeSteps = [];
+    model.readers.forEach((reader) => {
+      timeSteps.push(reader.timeStep);
+    });
+    return timeSteps.sort();
+  };
+
+  publicAPI.getTimeRange = () => {
+    const timeSteps = publicAPI.getTimeSteps();
+    const length = timeSteps.length;
+    if (length >= 1) {
+      return [timeSteps[0], timeSteps[length - 1]];
+    }
+    return [];
+  };
+
+  publicAPI.setUpdateTimeStep = (timeStep) => {
+    // TODO find a better way to check for timeSteps
```

Comment:
@finetjul what do you think would be the best approach here ? snap to the closest timeStep, only accept valid timesteps, use indices instead, something else ?

Reply:
I think the best is to have the same behavior as VTK C++ (which I do not know)

Reply:
VTK C++ seems to leave this choice to the reader implementation.

The UPDATE_TIME_STEP key usually returns a double value, so that leaves out the indices option.

I looked at vtkExtractTimeSteps, and it clamps the input value to the timestep range, then has a configurable option to either use the previous, next or nearest timestep.

For vtkGLTFReader, I think we chose to interpolate, because we could.

I think what makes sense here is to always snap to the previous value. We can do nearest instead, but that seems less consistent.

Reply:
I agree with you, snapping to the previous value seems to make more sense.

Reply:
@laurentmalka  please mark this conversation as resolved if you fixed it in your following work.